restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/ua/UserAgentManagerWithSBCTest.java

```diff
@@ -0,0 +1,331 @@
+        phone6 = sipStack6.createSipPhone("127.0.0.1", SipStack.PROTOCOL_TCP, restcommPort, aliceContact6);
+
+        sipStack7 = tool7.initializeSipStack(SipStack.PROTOCOL_TCP, "127.0.0.1", alicePort7, restcommContact);
+        phone7 = sipStack7.createSipPhone("127.0.0.1", SipStack.PROTOCOL_TCP, restcommPort, aliceContact7);
+    }
+
+    @After
+    public void after() throws Exception {
+        if (phone != null) {
+            phone.dispose();
+        }
+        if (sipStack != null) {
+            sipStack.dispose();
+        }
+        if (phone2 != null) {
+            phone2.dispose();
+        }
+        if (sipStack2 != null) {
+            sipStack2.dispose();
+        }
+        if (phone3 != null) {
+            phone3.dispose();
+        }
+        if (sipStack3 != null) {
+            sipStack3.dispose();
+        }
+        if (phone4 != null) {
+            phone4.dispose();
+        }
+        if (sipStack4 != null) {
+            sipStack4.dispose();
+        }
+        if (sipStack5 != null) {
+            sipStack5.dispose();
+        }
+//        deployer.undeploy("UserAgentTest");
+    }
+
+    /*
+     * Ensures that the OPTIONS comes back with the contact it was sent with in REGISTER
+     */
+    @Test
+    public void registerUserAgentWithOptionsPing() throws ParseException, InterruptedException {
+//        deployer.deploy("UserAgentTest");
+        // Register the phone so we can get OPTIONS pings from RestComm.
+        SipURI uri = sipStack.getAddressFactory().createSipURI(null, restcommContact);
+        Credential c = new Credential("127.0.0.1","alice", "1234");
+        phone.addUpdateCredential(c);
+        assertTrue(phone.register(uri, "alice", "1234", "sip:QfEEo7uZUiql5p84dw3l@127.0.0.1:" + alicePort + ";extraparam=restcomr0cks", 3600, 3600));
+        Thread.sleep(500);
+        assertTrue(MonitoringServiceTool.getInstance().getRegisteredUsers(deploymentUrl.toString(),adminAccountSid, adminAuthToken)==1);
+        // This is necessary for SipUnit to accept unsolicited requests.
+//        phone.setLoopback(true);
+        phone.listenRequestMessage();
+        RequestEvent requestEvent = phone.waitRequest(75000);
+        assertNotNull(requestEvent);
+        assertTrue(requestEvent.getRequest().getMethod().equals(SipRequest.OPTIONS));
+        assertEquals("sip:QfEEo7uZUiql5p84dw3l@127.0.0.1:" + alicePort + ";extraparam=restcomr0cks", requestEvent.getRequest().getRequestURI().toString());
+        logger.info("RequestEvent :"+requestEvent.getRequest().toString());
+        Response response = sipStack.getMessageFactory().createResponse(SIPResponse.OK, requestEvent.getRequest());
+        phone.sendReply(requestEvent, response);
+        Thread.sleep(1000);
+        // Clean up (Unregister).
+        assertTrue(phone.unregister(aliceContact, 0));
+    }
+
+    @Test
+    public void registerUserAgentWithExtraParamsAndOptionsPing() throws ParseException, InterruptedException {
```

Comment:
"alternate" category

Reply:
I copied the UserAgentManagerWithTest.java, not sure if it was present initially there